app/models/thread.php

```diff
@@ -61,4 +79,106 @@ public function create(Comment &$comment)
             $db->rollback();
         }
     }
-}
\ No newline at end of file
+
+    public function edit(Comment &$comment)
+    {
+        $this->validate();
+        $comment->validate();
+
+        if ($this->hasError() || $comment->hasError()) {
+            throw new ValidationException('Invalid thread or comment.');
+        }
+
+        $db = DB::conn();
+        $db->begin();
+        try {
+
+            $db->query("UPDATE thread SET title=?, category_name=?,
+                last_modified=NOW() WHERE id=?",
+                array($this->title, $this->category, $this->id)
+            );
+
+            $comment->edit();
+            $db->commit();
+        } catch (PDOException $e) {
+            $db->rollback();
+        }
+    }
+
+    public static function getAuthorById($id)
+    {
+        $db = DB::conn();
+        return $db->value("SELECT user_id FROM thread WHERE id=?", array($id));
+    }
+
+    public static function delete($thread_id)
+    {
+        $db = DB::conn();
+        $db->query("DELETE FROM thread WHERE id = ?", array($thread_id));
+    }
+
+    public function isAuthor($session_user)
+    {
+        return $this->user_id === $session_user;
+    }
+
+    public static function hasThread($user_id)
+    {
+        $db = DB::conn();
+        return $db->rows("SELECT id FROM thread WHERE user_id = ?",
+            array($user_id)
+        );
+    }
+
+    public static function getByUserId($user_id)
+    {
+        $db = DB::conn();
+        $threads = array();
+        $rows = $db->rows("SELECT * FROM thread WHERE user_id = ?",
+            array($user_id)
+        );
```

Comment:
Will change po, I just break this line po because It already reached 80th character po. :smile: